src/cow.rs

```diff
@@ -0,0 +1,85 @@
+#![cfg(feature = "alloc")]
+
+extern crate alloc;
+
+use crate::len_type::LenType;
+use crate::string::StringView;
+use crate::String;
+use core::borrow::Borrow;
+
+/// A clone-on-write (COW) string type for heapless.
+///
+/// `Cow` can either **borrow** a `StringView` or **own** a `String`.
+/// This allows efficient handling of strings that may be either temporary
+/// references or fully owned data.
+///
+/// # Type Parameters
+///
+/// - `N`: The inline buffer size for owned strings.
+/// - `LenT`: The integer type used for length tracking (must implement [`LenType`]).
+
+/// A clone-on-write string type for heapless
+#[derive(Debug)]
+pub enum Cow<'a, const N: usize, LenT: LenType = usize> {
```

Comment:
Unlike the `std::borrow::Cow`, this one is limited to strings only. I feel that naming it as such, would give the wrong impression. I'd rather we have a different name (e.g `CowStr`).